packages/react-storage/src/components/StorageBrowser/actions/handlers/listLocationsAction.ts

```diff
@@ -0,0 +1,42 @@
+import { Permission } from '@aws-amplify/storage/internals';
+
+import {
+  ActionInputConfig,
+  ListActionOptions,
+  ListActionInput,
+  ListActionOutput,
+  ListAction,
+  LocationType,
+} from '../types';
+
+export interface LocationData<T = Permission, K = LocationType> {
+  bucket: string;
+  permission: T;
+  prefix: string;
+  type: K;
+}
+
+export interface ListLocationsActionOptions<T = Permission | LocationType>
+  extends ListActionOptions {
+  exclude?: T | T[];
+}
+export interface ListLocationsActionInput<T = Permission | LocationType>
+  // `ListLocations` does not require `prefix`
+  extends Omit<
```

Comment:
Instead of omitting and redefining these types, can we create a base interface that doesn't include bucket and prefix, then extend it wherever needed?

Reply:
I think it's better to avoid using Omit if there's a more straightforward way to do this, mostly just for readability/clarity. I've also seen it cause type inference issues with complicated types like discriminated unions etc but that doesn't look like the case here.